scripts/asr_language_modeling/ngram_lm/opengrm.py

```diff
@@ -0,0 +1,224 @@
+
+        ps = subprocess.Popen(" ".join(sh_args), shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,)
+        return ps.communicate()[0]
+
+
+def perplexity(ngram_bin_path, ngram_mod, test_far):
+    sh_args = [
+        os.path.join(ngram_bin_path, "ngramperplexity"),
+        "--v=1",
+        ngram_mod,
+        test_far,
+        "--OOV_symbol=d",
+    ]
+    ps = subprocess.run(sh_args, capture_output=False, text=True, stdout=sys.stdout, stderr=sys.stderr)
+    return ps
+
+
+def make_arpa(ngram_bin_path, ngram_mod, ngram_arpa, force):
+    if os.path.isfile(ngram_arpa) and not force:
+        print("File", ngram_arpa, "exists. Skipping.")
+        return
+    else:
+        sh_args = [
+            os.path.join(ngram_bin_path, "ngramprint"),
+            "--ARPA",
+            ngram_mod,
+            ngram_arpa,
+        ]
+        return subprocess.run(sh_args, capture_output=False, text=True, stdout=sys.stdout, stderr=sys.stderr,)
+
+
+def make_kenlm(kenlm_bin_path, ngram_arpa, force):
+    ngram_kenlm = ngram_arpa + ".kenlm"
+    if os.path.isfile(ngram_kenlm) and not force:
+        print("File", ngram_kenlm, "exists. Skipping.")
+        return
+    else:
+        sh_args = [kenlm_bin_path, "trie", "-i", ngram_arpa, ngram_kenlm]
+        return subprocess.run(sh_args, capture_output=False, text=True, stdout=sys.stdout, stderr=sys.stderr,)
+
+
+def test_perplexity(ngram_bin_path, mod_c, symbols, test_txt, nemo_model_file, tmp_path):
+    test_far = os.path.join(tmp_path, os.path.split(test_txt)[1] + ".far")
+    farcompile(ngram_bin_path, symbols, test_txt, test_far, nemo_model_file, False, False, False)
+    res_p = perplexity(ngram_bin_path, mod_c, test_far)
+    return res_p
+
+
+def main(
+    ngram_bin_path, kenlm_bin_path, arpa_a, alpha, arpa_b, beta, out_path, test_file, symbols, nemo_model_file, force
+):
+
+    mod_c, arpa_c = merge(ngram_bin_path, arpa_a, alpha, arpa_b, beta, out_path, force)
+
+    if test_file and nemo_model_file:
+        if not symbols:
+            symbols = os.path.join(out_path, os.path.split(nemo_model_file)[1] + ".syms")
+            make_symbol_list(nemo_model_file, symbols, force)
+        test_p = test_perplexity(ngram_bin_path, mod_c, symbols, test_file, nemo_model_file, out_path)
+        print("Test perplexity", test_p)
+
+    print("Making ARPA model", arpa_c)
+    print("\n", make_arpa(ngram_bin_path, mod_c, arpa_c, force), "\n")
+    print("\n", make_kenlm(kenlm_bin_path, arpa_c, force), "\n")
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="Avg pytorch weights")
+    parser.add_argument(
+        "--ngram_bin_path",
+        required=False,
+        default="/root/miniconda3/bin",
+        type=str,
+        help="The path to the bin folder of Ngram",
+    )
+    parser.add_argument(
+        "--kenlm_bin_path",
+        required=False,
+        default="/workspace/nemo/decoders/kenlm/build/bin/build_binary",
```

Comment:
Let's drop the default and make sure users pass these arguments.

Reply:
Done